crypto/signature.go

```diff
@@ -25,6 +25,19 @@ func VerifySignedMessageByOwner(
 	}
 
 	if isEOASignature {
+		// EIP 155 https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md
+		if signature[64] == 27 || signature[64] == 28 {
+			//{0,1} + 27
+			signature[64] -= byte(27)
+		} else if signature[64] != 0 && signature[64] != 1 {
+			//{0,1} + CHAIN_ID * 2 + 35
+			chainID, err := client.NetworkID(context.Background())
+			if err != nil {
+				return err
+			}
+			signature[64] -= byte(chainID.Uint64()*2 + 35)
+		}
+
```

Comment:
First a question:
Are we totally sure that there is no support for EIP-155 in ethereum go-lib
Honestly I just spent 30 minutes looking for it... I saw many changes relating to it... but not anything specific to `SigToPub`.
It is just that I find it really strange that we have to write the code ourselves...

Just 2 things:
1. Maybe add a comment explaining we are changing the `v` parameter of the signature?
2. Will this code be imported from spec and used in production? If so, note that you will be making network calls for the client and this can be optimized. If it won't be used in production then no worries.